sdk/storage/azure-storage-file/azure/storage/file/models.py

```diff
@@ -745,3 +760,77 @@ def __str__(self):
 SharePermissions.LIST = SharePermissions(list=True) # type: ignore
 SharePermissions.READ = SharePermissions(read=True) # type: ignore
 SharePermissions.WRITE = SharePermissions(write=True) # type: ignore
+
+
+class NTFSAttributes(object):
+    """
+    Valid set of attributes to set for file or directory.
+    To set attribute for directory, 'Directory' should always be enabled except setting 'None' for directory.
+
+    :ivar bool read_only:
+        Enable/disable 'ReadOnly' attribute for DIRECTORY or FILE
+    :ivar bool hidden:
+        Enable/disable 'Hidden' attribute for DIRECTORY or FILE
+    :ivar bool system:
+        Enable/disable 'System' attribute for DIRECTORY or FILE
+    :ivar bool none:
+        Enable/disable 'None' attribute for DIRECTORY or FILE to clear all attributes of FILE/DIRECTORY
+    :ivar bool directory:
+        Enable/disable 'Directory' attribute for DIRECTORY
+    :ivar bool archive:
+        Enable/disable 'Archive' attribute for DIRECTORY or FILE
+    :ivar bool temporary:
+        Enable/disable 'Temporary' attribute for FILE
+    :ivar bool offline:
+        Enable/disable 'Offline' attribute for DIRECTORY or FILE
+    :ivar bool not_content_indexed:
+        Enable/disable 'NotContentIndexed' attribute for DIRECTORY or FILE
+    :ivar bool no_scrub_data:
+        Enable/disable 'NoScrubData' attribute for DIRECTORY or FILE
+    """
+    def __init__(self, read_only=False, hidden=False, system=False, none=False, directory=False, archive=False,
+                 temporary=False, offline=False, not_content_indexed=False, no_scrub_data=False, _str=None):
+        if not _str:
+            _str = ''
+        self.read_only = read_only or ('ReadOnly' in _str)
+        self.hidden = hidden or ('Hidden' in _str)
+        self.system = system or ('System' in _str)
+        self.none = none or ('None' in _str)
```

Comment:
In the method signatures you have lowercase "none" - does this matter? Or is the service case-insensitive?

Reply:
Yeah, the service is not case sensitive